chain/client/src/chunk_validation.rs

```diff
@@ -261,18 +267,76 @@ fn pre_validate_chunk_state_witness(
         )));
     }
 
+    let block = Chain::get_apply_chunk_block_context(
+        epoch_manager,
+        last_chunk_block.header(),
+        &store.get_previous_header(last_chunk_block.header())?,
+        true,
+    )?;
+    let epoch_id = epoch_manager.get_epoch_id(&block.block_hash)?;
+    let chunk_header = last_chunk_block.chunks().get(shard_id as usize).unwrap().clone();
+    let new_transactions = &state_witness.new_transactions;
+
+    // Verify that all proposed transactions are valid.
+    if !new_transactions.is_empty() {
+        let shard_uid = epoch_manager.shard_id_to_uid(shard_id, &epoch_id)?;
+        let chunk_extra =
```

Comment:
`ChunkExtra` is only available for nodes that track shard, so we should not rely on stateless validator having that available
instead you should use state root from `state_witness.chunk_header`